src/test/java/org/gridsuite/voltageinit/server/VoltageInitParametersTest.java

```diff
@@ -299,86 +296,80 @@ protected VoltageInitParametersInfos buildParametersUpdate() {
-        assertEquals(0., openReacParameters.getSpecificVoltageLimits().stream().filter(voltageLimitOverride -> "VLHV1".equals(voltageLimitOverride.getVoltageLevelId()) && VoltageLimitOverride.VoltageLimitType.LOW_VOLTAGE_LIMIT.equals(voltageLimitOverride.getVoltageLimitType())).findAny().get().getLimit());
+    private static Consumer<VoltageLimitOverride> assertVoltageLimitOverride(final String levelId, final VoltageLimitType limitType) {
+        return assertVoltageLimitOverride(levelId, limitType, null);
     }
 
-    @Test
-    public void testAddRestrictedVoltageLevelReport() {
-        Map<String, Double> restrictedVoltageLevel = new HashMap<>();
-        restrictedVoltageLevel.put("vl", 10.0);
-        ReporterModel reporter = new ReporterModel("test", "test");
-        VoltageInitWorkerService.addRestrictedVoltageLevelReport(restrictedVoltageLevel, reporter);
-        assertEquals("restrictedVoltageLevels", reporter.getReports().stream().findFirst().get().getReportKey());
-        assertEquals("The modifications to the low limits for certain voltage levels have been restricted to avoid negative voltage limits: vl : 10.0",
-                reporter.getReports().stream().findFirst().get().getDefaultMessage());
-        Optional<Map.Entry<String, TypedValue>> typedValues = reporter.getReports().stream()
-                .map(Report::getValues)
-                .findFirst()
-                .flatMap(values -> values.entrySet().stream().findFirst());
-        assertEquals("reportSeverity", typedValues.map(Map.Entry::getKey).get());
-        assertEquals("WARN", typedValues.map(value -> value.getValue().getValue()).get());
+    @SuppressWarnings("unchecked")
+    private static Consumer<VoltageLimitOverride> assertVoltageLimitOverride(final String levelId, final VoltageLimitType limitType, final Double limit) {
+        return voltageLimitOverride -> assertThat(voltageLimitOverride).as("voltageLimit override")
+            .is(nestable("VoltageLimitOverride", Stream.<Condition<VoltageLimitOverride>>of(
+            verboseCondition(actual -> levelId.equals(actual.getVoltageLevelId()),
+                "to have voltageLevelId=\"" + Objects.toString(levelId, "<null>") + "\"",
+                actual -> " but is actually \"" + Objects.toString(actual.getVoltageLevelId(), "<null>") + "\""),
+            verboseCondition(actual -> limitType.equals(actual.getVoltageLimitType()),
+                "to have voltageLimitType=\"" + Objects.toString(limitType, "<null>") + "\"",
+                actual -> " but is actually \"" + Objects.toString(actual.getVoltageLimitType(), "<null>") + "\""),
+            limit == null ? null : verboseCondition(actual -> limit.equals(actual.getLimit()),
+                "to have limit=" + limit, actual -> " but is actually " + actual.getLimit())
+        ).filter(Objects::nonNull).toArray(Condition[]::new)));
     }
-}
 
+    @TestFactory
+    List<DynamicTest> dynamicTestsBuildSpecificVoltageLimits() {
```

Comment:
Do not use DynamicTest. We can't use different ways of writing UT in every repos